### PR TITLE
[book] Document NoteCommit region optimisations

### PR DESCRIPTION
**Pull request by therealyingtong**
Documents changes made in #169.

### CODE REVIEW

File: book/src/design/circuit/gadgets/sinsemilla/note-commit.md [hunk 1]
```diff
@@ -423,28 +433,100 @@ In these cases, we check that $\psi_{0..=253} < t_\mathbb{P}$:
...
 \end{array}
 $$
+
+Then, to check that the decomposition was correct:
+$$
+\begin{array}{|c|l|}
+\hline
+\text{Degree} & \text{Constraint} \\\hline
+2 & q_{\NoteCommit,3} \cdot \left(j - (\textsf{LSB} + k_0 \cdot 2 + k_1 \cdot 2^{10}) \right) = 0 \\\hline
+2 & q_{\NoteCommit,3} \cdot \left(y - (j + k_2 \cdot 2^{250} + k_3 \cdot 2^{254}) \right) = 0 \\\hline
```
Comment (therealyingtong) [inline book/src/design/circuit/gadgets/sinsemilla/note-commit.md:492]:
```suggestion
2 & q_{\textsf{y\_canon}} \cdot \left(j - (\textsf{LSB} + k_0 \cdot 2 + k_1 \cdot 2^{10}) \right) = 0 \\\hline
2 & q_{\textsf{y\_canon}} \cdot \left(y - (j + k_2 \cdot 2^{250} + k_3 \cdot 2^{254}) \right) = 0 \\\hline
```

File: book/src/design/circuit/gadgets/sinsemilla/note-commit.md [hunk 2]
```diff
@@ -423,28 +433,100 @@ In these cases, we check that $\psi_{0..=253} < t_\mathbb{P}$:
...
-    b     &        d       &        e       &   g    &     h    &        d_1        &  x(pk_d) &      b_3       &     a'   &   b_2     &        0          &         0         \\\hline
-  {e_1}f' &  {g_1}{g_2}'   &        v       &  d_2   &  z_{d,1} &        e_0        &{b_3}{c}' &       c        &     a    &  x(g_d)   &        1          &         0         \\\hline
-   e_1    &        f       &       g_0      &  g_1   &  z_{g,1} &        h_0        &    h_1   &      d_0       &    b_0   &   b_1     &        0          &         1         \\\hline
-  \rho    &    z_{f,13}    & z_{{e_1}f',14} &  \psi  & z_{g,13} &z_{{g_1}{g_2}',13} & z_{c,13} & z_{{b_3}c',14} & z_{a,13} & z_{a',13} &        0          &         0         \\\hline
+\begin{array}{|c|l|}
+\hline
+\text{Degree} & \text{Constraint} \\\hline
+  & \ShortLookupRangeCheck{k_0, 9} \\\hline
+  & \ShortLookupRangeCheck{k_2, 4} \\\hline
+3 & q_{\NoteCommit,3} \cdot \BoolCheck{k_3} = 0 \\\hline
```
Comment (therealyingtong) [inline book/src/design/circuit/gadgets/sinsemilla/note-commit.md:482]:
```suggestion
3 & q_{\textsf{y\_canon}} \cdot \BoolCheck{k_3} = 0 \\\hline
```

File: book/src/design/circuit/gadgets/sinsemilla/note-commit.md [hunk 3]
```diff
@@ -423,28 +433,100 @@ In these cases, we check that $\psi_{0..=253} < t_\mathbb{P}$:
...
+       $$k_3 \cdot z_{j',13} = 0.$$
+
+$$
+\begin{array}{|c|l|}
+\hline
+\text{Degree} & \text{Constraint} \\\hline
+3 & q_{\NoteCommit,3} \cdot k_3 \cdot k_2 = 0 \\\hline
+3 & q_{\NoteCommit,3} \cdot k_3 \cdot z_{j,13} = 0 \\\hline
+2 & q_{\NoteCommit,3} \cdot (j + 2^{130} - t_\mathbb{P} - j') = 0 \\\hline
+3 & q_{\NoteCommit,3} \cdot k_3 \cdot z_{j',13} = 0 \\\hline
```
Comment (therealyingtong) [inline book/src/design/circuit/gadgets/sinsemilla/note-commit.md:527]:
```suggestion
3 & q_{\textsf{y\_canon}} \cdot k_3 \cdot k_2 = 0 \\\hline
3 & q_{\textsf{y\_canon}} \cdot k_3 \cdot z_{j,13} = 0 \\\hline
2 & q_{\textsf{y\_canon}} \cdot (j + 2^{130} - t_\mathbb{P} - j') = 0 \\\hline
3 & q_{\textsf{y\_canon}} \cdot k_3 \cdot z_{j',13} = 0 \\\hline
```

**Review by daira — APPROVED** [review]
utACK. Checked for consistency with #169.

**Review by str4d — APPROVED** [review]

**Pull request merged**